### PR TITLE
Automatic range on insights graph

### PR DESCRIPTION
**Pull request by HuldaCZ**
## Description
We where missing option to show specific range of data on insights time-series graph. Not just $NOW() - XXX. 
In this PR is implementation of Automatic range based on data inserted to time-series. So you can specify any time range in filter. 

In case there is no filter all data in collection will appear. 

Related topics in discussions are: 
[https://github.com/directus/directus/discussions/14712](https://github.com/directus/directus/discussions/14712)
[https://github.com/directus/directus/discussions/12945](https://github.com/directus/directus/discussions/12945)
[https://github.com/directus/directus/discussions/10619](https://github.com/directus/directus/discussions/10619)

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [ ] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated

### CODE REVIEW

File: app/src/panels/time-series/index.ts [hunk 1]
```diff
@@ -12,28 +12,39 @@ export default definePanel({
...
+					{
+						[options.dateField]: {
+							_lte: `$NOW`,
+						},
+					},
+					options.filter || {},
+				],
+			};
+		};
+
```
Comment (azrikahar) [inline app/src/panels/time-series/index.ts:39]: Maybe we can change this to something like this:

```js
		const filter: Filter = {
			_and: [options.filter || {}],
		};

		if (options.range !== 'auto') {
			filter._and.push(
				{
					[options.dateField]: {
						_gte: `$NOW(-${options.range || '1 week'})`,
					},
				},
				{
					[options.dateField]: {
						_lte: `$NOW`,
					},
				}
			);
		}
```	

understandably your code would prevent the unnecessary `_and` wrapper when it's auto, but this suggestion is more so to align with the existing similar "push to filter" logics using throughout the app, just for consistency sake. Examples such as:

https://github.com/directus/directus/blob/f77d12eff4b11862415a432ddf3be97a4d4a729e/app/src/composables/use-revisions.ts#L37-L60

https://github.com/directus/directus/blob/f77d12eff4b11862415a432ddf3be97a4d4a729e/app/src/modules/files/routes/collection.vue#L257-L281

Comment (HuldaCZ) [reply app/src/panels/time-series/index.ts:39]: Sure, it's look cleaner and more consistent. I will update PR with this change.

File: app/src/panels/time-series/panel-time-series.vue [hunk 2]
```diff
@@ -116,6 +116,15 @@ function setupChart() {
 
 	const isFieldTimestamp = fieldsStore.getField(props.collection, props.dateField)?.type === 'timestamp';
 
+	const allDates = props.data.map((metric) => {
+		return (
+			new Date(toISO(metric.group)).getTime() - (isFieldTimestamp ? new Date().getTimezoneOffset() * 60 * 1000 : 0)
```
[the remark this answers is not in the document]
Comment (HuldaCZ) [reply app/src/panels/time-series/panel-time-series.vue]: I just committed update, where I put this section of code to separate function called toIncludeTimezoneOffest. To make clear what the function do. Thank You for code review.

File: app/src/panels/time-series/index.ts [hunk 3]
```diff
@@ -12,28 +13,33 @@ export default definePanel({
...
-						},
-						{
-							[options.dateField]: {
-								_lte: `$NOW`,
-							},
-						},
-						options.filter || {},
-					],
-				},
+				filter: filter,
```
Comment (azrikahar) [inline app/src/panels/time-series/index.ts:42]:
```suggestion
				filter,
```

File: app/src/panels/time-series/panel-time-series.vue [hunk 4]
```diff
@@ -116,10 +116,16 @@ function setupChart() {
 
 	const isFieldTimestamp = fieldsStore.getField(props.collection, props.dateField)?.type === 'timestamp';
 
+	const allDates = props.data.map((metric) => {
+		return toIncludeTimezoneOffest(metric.group, isFieldTimestamp);
```
Comment (azrikahar) [inline app/src/panels/time-series/panel-time-series.vue:120]:
```suggestion
		return toIncludeTimezoneOffset(metric.group, isFieldTimestamp);
```

File: app/src/panels/time-series/panel-time-series.vue [hunk 5]
```diff
@@ -116,10 +116,16 @@ function setupChart() {
...
+	});
+
+	const minDate = Math.min(...allDates);
+	const maxDate = Math.max(...allDates);
+
 	metrics.value = orderBy(
 		props.data.map((metric) => ({
-			x: new Date(toISO(metric.group)).getTime() - (isFieldTimestamp ? new Date().getTimezoneOffset() * 60 * 1000 : 0),
-
+			x: toIncludeTimezoneOffest(metric.group, isFieldTimestamp),
```
Comment (azrikahar) [inline app/src/panels/time-series/panel-time-series.vue:128]:
```suggestion
			x: toIncludeTimezoneOffset(metric.group, isFieldTimestamp),
```

File: app/src/panels/time-series/panel-time-series.vue [hunk 6]
```diff
@@ -289,6 +298,10 @@ function setupChart() {
 
 	chart.value.render();
 
+	function toIncludeTimezoneOffest(time: Record<string, any>, isFieldTimestamp: boolean): number {
```
Comment (azrikahar) [inline app/src/panels/time-series/panel-time-series.vue:301]:
```suggestion
	function toIncludeTimezoneOffset(time: Record<string, any>, isFieldTimestamp: boolean): number {
```

**Review by azrikahar — APPROVED** [review]
LGTM! Thanks for your first contribution here 😊

**Pull request merged**